### DISCUSSION TITLE
How to add click event to the button

### DISCUSSION THREAD

**Discussion by yashwantnmane** [GrapesJS/grapesjs#4850] [Q&A] [answered]
Hello @artf,

I would like to thank you for this library. It helped me a lot.

I'm using this to build a website for page development, I'm having an issue adding a click event on a button component definition.

{
  type: 'button',
  attributes: {
    class: 'btn btn-carousel btn-carousel-prev',
    title: 'Prev',
  },
  text: '<span>&#10094;</span>',
  styles: `.btn-carousel-prev {
    left: 10px;
  }`
}

above is my button component and I want to do some stuff on the click. Please help to achieve this.

**Comment by allenhwkim** [comment, marked as answer]
I see from this document to set `view` - `events` for a type, https://grapesjs.com/docs/modules/Components.html#define-custom-component-type
```
  type: 'button',
  ...
  view: {
    events: {
      'click': e => console.log(`button clicked`)
    },
   ...
```

**Reply by yashwantnmane** [reply]
Thank you, Allen, it worked.
